code/core/src/manager/hooks/useModalDialog.ts

```diff
@@ -0,0 +1,43 @@
+import { useEffect, useRef } from 'react';
+
+interface UseModalDialogOptions {
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+export function useModalDialog({ isOpen, onClose }: UseModalDialogOptions) {
+  const dialogRef = useRef<HTMLDialogElement>(null);
+
+  useEffect(() => {
+    const dialogNode = dialogRef.current;
+    if (dialogNode) {
+      if (isOpen) {
+        if (!dialogNode.hasAttribute('open')) {
+          dialogNode.showModal();
+        }
+      } else {
+        if (dialogNode.hasAttribute('open')) {
+          dialogNode.close();
+        }
```

Comment:
This might be where you need to do something to trigger unmounting the Transition and letting the Transition do the closing for you. Which ironically might mean we need to preserve the `Esc` handler for this use case :( 

We need to go through this mess because Firefox does not yet support `allow-discrete` that would let us use native CSS to animate the exit animation of the dialog (see https://frontendmasters.com/blog/the-dialog-element-with-entry-and-exit-animations/ and https://codepen.io/sidnioulz/pen/PwqNQzE for a simple relevant codepen).

@ghengeveld your thoughts here on how to proceed would be appreciated.